src/main/java/pl/zaprogramujzycie/mma/controllers/MedicineController.java

```diff
@@ -64,8 +65,13 @@ ResponseEntity<MedicineResponse> createMedicine(@AuthenticationPrincipal final P
             @ApiResponse(responseCode = "500", description = "Internal server error")
     })
     @GetMapping("/{id}")
-    ResponseEntity<MedicineResponse> findById(@AuthenticationPrincipal final Principal principal, @PathVariable final long id) {
-            return null;
+    ResponseEntity<MedicineResponse> findById(final Principal principal, @PathVariable final long id) {
+        MedicineResponse response = service.findById(id, principal);
+        if (response != null) {
+            return ResponseEntity.ok(response);
```

Comment:
Obsluga bledow nie powinna sie znajdowac w kontrolerze, rzuc wyjatek w serwisie i zmapuje go na kod bledu za pomoc @RestControllerAdvice

Reply:
Nawet mam ten plik @RestControllerAdvice przygotowany. Ale mam tak wbite do głowy z Javy EE, że wyjątki należy obsłużyć za pomocą bloku try-catch, że mam opory przed wypuszczeniem błędu "w świat" niech go coś tam ogarnie, mimo, że wiem że ogarnie :) Już poprawiłam, ale jak wrzucę to jeszcze zerknij, bo ja tak na 100% to nie jestem pewna, które mam rzucać, a które nie.